pkg/registry/batch/job/strategy.go

```diff
@@ -100,6 +100,11 @@ func (jobStrategy) PrepareForCreate(ctx context.Context, obj runtime.Object) {
 		job.Spec.PodFailurePolicy = nil
 	}
 
+	if !utilfeature.DefaultFeatureGate.Enabled(features.JobBackoffLimitPerIndex) {
```

Comment:
We also need to drop pod failure policies that use `FailIndex` when the feature is disabled.

Reply:
Well spot, added. 
FYI @deads2k as this is an api strategy change.